docs/book/getting_started/core-concepts.md

```diff
@@ -0,0 +1,151 @@
+---
+description: A good place to start before diving further into the docs.
+---
+
+# Core Concepts
+
+ZenML consists of a few components. This guide walks through the various pieces you'll encounter, starting from the basics to things you'll only encounter when deploying to the cloud. 
+
+Here's a high-level overview of a typical ZenML workflow:
+
+1. Writing a *pipeline* to define what happens in your machine learning workflow.
+2. Configuring a ZenML *stack*.
+3. Switching between *stacks* depending on needs.
+4. Customizing your *stack* with different *components*.
+
+So first, the basics.
+
+## Pipeline and Steps
+
+At its core, ZenML follows a pipeline-based workflow for your data projects.
+A **pipeline** consist of a series of **steps**, organized in any order that makes sense for your use case. 
+
+Below, you can see three **steps** running one after another in a **pipeline**. 
+
+![The most basic ZenML pipeline](../assets/core_concepts/concepts-1.png)
+
+The steps might have dependencies between them. 
+For example, a step might use the outputs from a previous step and thus must wait until the previous step completes before starting. This is something you can keep in mind when organizing your steps.
+
+You can have multiple pipelines for different purposes. For example, a training pipeline to train and evaluate models and an inference pipeline to serve the model.
+
+Pipelines and steps are defined in code using Python *decorators*.
```

Comment:
A class based API is available too for advanced use-cases. Maybe say that using Python *decorators* or *classes*

As a bonus, maybe link to what decorators are